### PR TITLE
build: bump libevent to c4fbae3a.

### PR DESCRIPTION
**Pull request by htuch**
This brings in buffer correctness fixes for https://github.com/libevent/libevent/issues/774 and https://github.com/libevent/libevent/issues/778.

This will temporarily take us away from a released version until 2.1.9.

Fixes oss-fuzz issue https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=13400.

Risk level: Low (since we're already at 2.1.9-beta, which is pretty near in history).
Testing: regression unit tests added upstream in libevent, corpus entry added.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
Tagging https://github.com/libevent/libevent/issues/778

File: test/common/buffer/event_buffer_test.cc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  // special case handling for empty, unlike evbuffer_prepend_buffer.
+  evbuffer_prepend(buf, "", 0);
+
+  // Transfer a byte of buf to other_buf.
+  int rc = evbuffer_remove_buffer(buf, other_buf, 1);
+  ASSERT_EQ(rc, 1);
+
+  // Try and append "bar" to buf. This prepends, rather than appends!
+  evbuffer_add(buf, "bar", 3);
+  const char* s = reinterpret_cast<char*>(evbuffer_pullup(buf, -1));
```
Comment (lizan) [inline test/common/buffer/event_buffer_test.cc]: Is `s` guaranteed to be NUL-terminated? same below.

Comment (htuch) [reply test/common/buffer/event_buffer_test.cc]: No, you're right, I should probably use buffer length and form a `std::string`. OTOH, a good question is do we need these cases. `libevent` has added regressions itself, so we could just nuke this entire test; I wrote it originally to share reproducers with `libevent` folks.

Comment (mattklein123) [reply test/common/buffer/event_buffer_test.cc]: Yeah I would probably just kill it if they have test coverage for this now.

**Pull request merged**